hack/jenkins/test-flake-chart/report_flakes/report_flake.go

```diff
@@ -0,0 +1,256 @@
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/storage"
+)
+
+type shortSummary struct {
+	NumberOfTests int
+	NumberOfFail  int
+	NumberOfPass  int
+	NumberOfSkip  int
+	FailedTests   []string
+	PassedTests   []string
+	SkippedTests  []string
+	Durations     map[string]float64
+	TotalDuration float64
+	GopoghVersion string
+	GopoghBuild   string
+	Detail        struct {
+		Name     string
+		Details  string
+		PR       string
+		RepoName string
+	}
+}
+
+// parseEnvironmentList reads the existing environments from the file
+func parseEnvironmentList(listFile string) ([]string, error) {
+	data, err := os.ReadFile(listFile)
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.TrimSpace(string(data)), "\n"), nil
+}
+
+func testSummariesFromGCP(ctx context.Context, pr, rootJob string, envList []string, client *storage.Client) (map[string]*shortSummary, error) {
+	envToSummaries := map[string]*shortSummary{}
+	for _, env := range envList {
+		summary, err := testSummaryFromGCP(ctx, pr, rootJob, env, client)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch %s test summary from gcp, err: %v", env, err)
+		}
+		if summary != nil {
+			// if the summary is nil(missing) we just skip it
+			envToSummaries[env] = summary
+		}
+	}
+	return envToSummaries, nil
+}
+
+// testSummaryFromGCP gets the summary of a test for the specified env.
+func testSummaryFromGCP(ctx context.Context, pr, rootJob, env string, client *storage.Client) (*shortSummary, error) {
+
+	btk := client.Bucket("minikube-builds")
+	obj := btk.Object(fmt.Sprintf("logs/%s/%s/%s_summary.json", pr, rootJob, env))
+
+	reader, err := obj.NewReader(ctx)
+	if errors.Is(err, storage.ErrObjectNotExist) {
+		// if this file does not exist, just skip it
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	// read the file
+	data, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	var summary shortSummary
+	if err = json.Unmarshal(data, &summary); err != nil {
+		return nil, fmt.Errorf("failed to deserialize the file: %v", err)
+	}
+	return &summary, nil
+
+}
+
+// flakeRate downloads recent flake rates from GCS, and returns a map{env->map{testname->flake rate}}
+func flakeRate(ctx context.Context, client *storage.Client) (map[string]map[string]float64, error) {
+	btk := client.Bucket("minikube-flake-rate")
+	obj := btk.Object("flake_rates.csv")
+	reader, err := obj.NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read the flake rate file: %v", err)
+	}
+	// parse the csv file to the map
+	records, err := csv.NewReader(reader).ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read and parse the flake rate file: %v", err)
+	}
+	result := map[string]map[string]float64{}
+	for i := 1; i < len(records); i++ {
+		// for each line in csv we extract env, test name and flake rate
+		if len(records[i]) < 2 {
```

Comment:
```suggestion
		if len(records[i]) < 3 {
```
`records[i][2]` is below, so need to increase length check